dfinity/ic
# feat(boundary): rate-limiting canister

This is an initial prototype of the future rate-limiting canister. So far this canister has only a single job of periodically quering the registry canister for the existing API boundary nodes.

**How to test**:
- Start a `system-test`. Note: it's better to add a `boundary-node` to the test `setup()`, as this makes canister installation with `dfx` easier:
```
$ ict test //rs/tests/boundary_nodes:api_bn_decentralization_test_head_nns --keepalive
```
- Deploy the canister:
```
$ dfx deploy rate_limiting --network https://ic0.farm.dfinity.systems --no-wallet
```
- Check the API boundary nodes count with either `query` or `update`:
```
$ dfx canister call --network https://ic0.farm.dfinity.systems qsgjb-riaaa-aaaaa-aaaga-cai get_api_boundary_nodes_count --query
(2 : nat64)
```
